modules/core/shared/src/main/scala/io/circe/JsonNumber.scala

```diff
@@ -236,6 +237,15 @@ private[circe] final case class JsonFloat(value: Float) extends JsonNumber {
  */
 object JsonNumber {
 
+  /**
+  * Constant pool of integer numbers between -128 to 127, similar to the JVM's constant pool for integers.
+  */
+  private val jsonLongConstantPool: ArraySeq[JsonLong] =
+    ArraySeq.tabulate(255)(x => JsonLong((x - 128).toLong))
+
+  private[circe] final def fromLong(value: Long): JsonLong =
+    jsonLongConstantPool.applyOrElse[Int, JsonLong]((value - 128).toInt, _ => JsonLong(value))
```

Comment:
This is allocating a lambda `_ => JsonLong(value)` on every invocation. If the goal is performance, we might as well go for a raw `Array` and `if`/`else`.

Reply:
Nice catch
Also, turns out that function was completely wrong and was fetching the wrong numbers, not sure how I didn't notice in my tests 🙃
On top of that, it was boxing/unboxing unnecessarily.

Fixed in ae1166a

Decompiled code:
```java
    public final JsonLong fromLong(long value) {
        long idx = value + 128L;
        if (idx >= 0L && idx < (long)this.jsonLongConstantPool().size()) {
            return this.jsonLongConstantPool().apply((int)idx);
        }
        return new JsonLong(value);
    }
```

Reply:
Updated after results:
![image](https://github.com/user-attachments/assets/e3f60e28-8113-4c07-ad1f-7935cc5eaacf)
![image](https://github.com/user-attachments/assets/b58b8aa8-d83f-4c24-9df8-b2d3f0e4a559)

The number of JsonLong's is now much lower, as expected.
I have to admit that I'm a bit confused with the retained size increase, though... especially of JString (which should have been unaffected by this commit)... Maybe there's some non-determinism there that I'm not accounting for? 🤷